modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -24,15 +24,18 @@ import * as InsertList from './InsertList';
 
 const wrappedElements = [ 'pre' ];
 
-const shouldPasteContentOnly = (fragment: AstNode, parentNode: Element) => {
-  const firstNode = fragment.firstChild;
-
-  const isAFlattenableTag = Arr.contains(wrappedElements, firstNode.name);
-  const isPastingInTheSameTag = firstNode.name === parentNode.tagName.toLowerCase();
-  const lastNode = fragment.lastChild.attr('data-mce-type') === 'bookmark' ? fragment.lastChild.prev : fragment.lastChild;
-  const isCopingOnlyOneTag = firstNode === lastNode;
-
-  return isCopingOnlyOneTag && isAFlattenableTag && isPastingInTheSameTag;
+const shouldPasteContentOnly = (dom: DOMUtils, fragment: AstNode, parentNode: Element, editor: Editor): boolean => {
+  const firstNode = fragment.firstChild as AstNode;
+  const lastNode = fragment.lastChild as AstNode;
+  const last = lastNode.attr('data-mce-type') === 'bookmark' ? lastNode.prev : lastNode;
+
+  const isCopyingOnlyOneTag = firstNode === last;
+  const isWrappedElement = Arr.contains(wrappedElements, firstNode.name);
+  const isPastingInTheSameTag = firstNode.name === parentNode.nodeName.toLowerCase() ? true : dom.getParent(parentNode, (node) => firstNode.name === node.nodeName.toLowerCase()) !== null;
+  const isSameClass = firstNode.attr('class') !== undefined ? firstNode.attr('class') === parentNode.className : parentNode.className === '';
```

Comment:
Oh actually... was this added to exclude merging a codesample block? If so then yeah we might need to leave this as is for now, but either way we should have a test for it.

Also the change that I suggested in Lorenzo's [PR](https://github.com/tinymce/tinymce/pull/7996) would likely fix this without needing this check since it would then be marked as a CEF element while processing here. So maybe we need to make sure they are both merged for 6.1.1 🤔

Reply:
isSameClass to be removed once change from TINY-8861 is implemented, attaching `contenteditable="false"` attribute to codesample elements. This will remove the requirement to match class for codesample elements, and instead prevent unwrap since `isContentEditable` will be false.

Reply:
Just adding an action item here so we can't forget this.

- [ ] Either revert this once the other PR is merged or add test cases for this